src/libraries/Microsoft.Extensions.Caching.Memory/src/MemoryCache.cs

```diff
@@ -295,6 +313,88 @@ void ScheduleTask(DateTimeOffset utcNow)
             }
         }
 
+        internal readonly List<WeakReference<Stats>?> _allStats = new();
+        internal readonly Stats _accumulatedStats;
+
+        private (int, int) Sum()
+        {
+            int hits = 0, misses = 0;
+            lock (_allStats)
+            {
+                hits += _accumulatedStats.Hits;
+                misses += _accumulatedStats.Misses;
+                foreach (WeakReference<Stats>? wr in _allStats)
+                {
+                    if (wr is not null && wr.TryGetTarget(out Stats? stats))
+                    {
+                        hits += stats.Hits;
+                        misses += stats.Misses;
+                    }
+                }
+            }
+            return (hits, misses);
+        }
+
+        private ThreadLocal<Stats>? _stats;
+
+        private void Hit()
+        {
+            GetStats().Hits++;
+        }
+
+        private void Miss()
+        {
+            GetStats().Misses++;
+        }
+
+        private Stats GetStats()
+        {
+            if (_stats == null)
```

Comment:
This initialization could be done eagerly in the constructor and it would probably give a slight performance improvement. You would also need to move the code that appends to the _allStats list inside the init function passed to the ThreadLocal.ctor so that it only runs when the per-thread state is created.